### ISSUE TITLE
Changing the cache_size in Gemmlowp/meta/single_thread_gemm.h cause random error in Requantize nodes

### ISSUE THREAD

**Issue by qjivy** [tensorflow/tensorflow#10488]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
No

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Linux ubuntu 16.04

- **TensorFlow installed from (source or binary)**:

source, NDK built Android ARM64 binary

- **TensorFlow version (use command below)**:

commit id: f48673b5054b474fa1e51823edd075088cd16d5f
Author: Luke Iwanski <luke@codeplay.com>

- **Bazel version (if compiling from source)**:
0.4.5

- **CUDA/cuDNN version**:
no

- **GPU model and memory**:
no

- **Exact command to reproduce**:
1. bazel --output_base=../out/armv8_benchmark_model/ build -s -c opt --jobs=1 --crosstool_top=//external:android/crosstool --cpu=arm64-v8a --host_crosstool_top=@bazel_tools//tools/cpp:toolchain  tensorflow/tools/benchmark:benchmark_model 2>&1 |tee log.txt
2.adb push benchmark_model /data/local/tmp/
 
3. run_vgg1 on Nexus5X6P phone:
script:
time ./$1 --graph=$2   --input_layer="images:0"   --input_layer_shape="1,224,224,3"   --input_layer_type="float"   --output_layer="prob:0" --num_runs=1

The final command line that I use:
./run_vgg1.sh benchmark_model vgg16.8bit.weightsnodes.model

### My problem
When I change the cache_size from 256*1024 to 128*1024 in the Gemmlowp/meta/single_thread_gemm.h, the benchmark_model randomly failed on the 8 bit quantized both node and weights vgg16 model.

### Source code / logs
My Tensorflow commit-id:
commit f48673b5054b474fa1e51823edd075088cd16d5f

My modify for the Gemmlowp:
in file  gemmlowp/meta/single_thread_gemm.h:
change all the "int cache_size = 256 * 1024" to "int cache_size = 128 * 1024".

The error log is :
native : benchmark_model.cc:381 Graph: [vgg16.8bit.weightsnodes.model]
native : benchmark_model.cc:382 Input layers: [images:0]
native : benchmark_model.cc:383 Input shapes: [1,224,224,3]
native : benchmark_model.cc:384 Input types: [float]
native : benchmark_model.cc:385 Output layers: [prob:0]
native : benchmark_model.cc:386 Num runs: [1]
native : benchmark_model.cc:387 Inter-run delay (seconds): [-1.0]
native : benchmark_model.cc:388 Num threads: [-1]
native : benchmark_model.cc:389 Benchmark name: []
native : benchmark_model.cc:390 Output prefix: []
native : benchmark_model.cc:391 Show sizes: [0]
native : benchmark_model.cc:392 Warmup runs: [2]
native : benchmark_model.cc:52 Loading TensorFlow.
native : benchmark_model.cc:59 Got config, 0 devices
can't determine number of CPU cores: assuming 4
can't determine number of CPU cores: assuming 4
native : benchmark_model.cc:257 Running benchmark for 2 iterations without detailed stat logging:
native : benchmark_model.cc:233 Error during inference: Invalid argument: requested_output_max must be >= requested_output_min, but got nan and 0
	 [[Node: fc8/BiasAdd/eightbit/requantize = Requantize[Tinput=DT_QINT32, out_type=DT_QUINT8, _device="/job:localhost/replica:0/task:0/cpu:0"](fc8/BiasAdd/eightbit, fc8/BiasAdd/eightbit:1, fc8/BiasAdd/eightbit:2, fc8/BiasAdd/eightbit/requant_range, fc8/BiasAdd/eightbit/requant_range:1)]]
native : benchmark_model.cc:268 Failed on run 0
native : benchmark_model.cc:451 Timing failed with Invalid argument: requested_output_max must be >= requested_output_min, but got nan and 0
	 [[Node: fc8/BiasAdd/eightbit/requantize = Requantize[Tinput=DT_QINT32, out_type=DT_QUINT8, _device="/job:localhost/replica:0/task:0/cpu:0"](fc8/BiasAdd/eightbit, fc8/BiasAdd/eightbit:1, fc8/BiasAdd/eightbit:2, fc8/BiasAdd/eightbit/requant_range, fc8/BiasAdd/eightbit/requant_range:1)]]
    0m11.12s real     0m27.55s user     0m01.87s system

**Comment by petewarden** [comment]
Maciek, can you take a look at this?

**Comment by qjivy** [comment]
The same error seems happens on the quantized MobileNet_v1_1.0_224 too.
By reference this page:
https://github.com/tensorflow/models/tree/master/slim

**Using the following command line, I get the mobilenet_v1_224.pb.**

$ python export_inference_graph.py \
  --alsologtostderr \
  --model_name=mobilenet_v1 \
  --image_size=224 \
  --output_file=/tmp/mobilenet_v1_224.pb

**Then I use the following command line to freeze it.**

python bazel-bin/tensorflow/python/tools/freeze_graph --input_graph=../model/models/slim/download_models/mobilenet/mobilenet_v1_224.pb  --input_checkpoint=../model/models/slim/download_models/mobilenet/mobilenet_v1_1.0_224.ckpt --input_binary=true --output_graph=../model/models/slim/download_models/mobilenet/freeze_mobilenet_v1_224.pb --output_node_names=MobilenetV1/Predictions/Reshape_1

**Then quantized it using the "trasnform_graph" to quantize both the wights and node.
./transform_graph --in_graph=./freeze_mobilenet_v1_224.pb --out_graph=./quantized_freeze_mobilenet_v1_224.pb --inputs='input' --outputs="MobilenetV1/Predictions/Reshape_1" --transforms='quantize_weights quantize_nodes' 

**After** I got the quantized model, I run it with benchmark_model then get the fault.
command line of benchmark_model:
 ./benchmark_model_armv8a_0628 --graph=quantized_freeze_mobilenet_v1_224.pb --input_layer="input"   --input_layer_shape="1,224,224,3"   --input_layer_type="float"   --output_layer="MobilenetV1/Predictions/Reshape_1" --num_runs=20


---The error log---
native : benchmark_model.cc:382 Graph: [quantized_freeze_mobilenet_v1_224.pb]
native : benchmark_model.cc:383 Input layers: [input]
native : benchmark_model.cc:384 Input shapes: [1,224,224,3]
native : benchmark_model.cc:385 Input types: [float]
native : benchmark_model.cc:386 Output layers: [MobilenetV1/Predictions/Reshape_1]
native : benchmark_model.cc:387 Num runs: [20]
native : benchmark_model.cc:388 Inter-run delay (seconds): [-1.0]
native : benchmark_model.cc:389 Num threads: [-1]
native : benchmark_model.cc:390 Benchmark name: []
native : benchmark_model.cc:391 Output prefix: []
native : benchmark_model.cc:392 Show sizes: [0]
native : benchmark_model.cc:393 Warmup runs: [2]
native : benchmark_model.cc:53 Loading TensorFlow.
native : benchmark_model.cc:60 Got config, 0 devices
can't determine number of CPU cores: assuming 4
can't determine number of CPU cores: assuming 4
native : benchmark_model.cc:258 Running benchmark for 2 iterations without detailed stat logging:
native : benchmark_model.cc:234 Error during inference: Invalid argument: requested_output_max must be >= requested_output_min, but got nan and 0
	 [[Node: MobilenetV1/MobilenetV1/Conv2d_2_pointwise/BatchNorm/batchnorm/mul/eightbit/requantize = Requantize[Tinput=DT_QINT32, out_type=DT_QUINT8, _device="/job:localhost/replica:0/task:0/cpu:0"](MobilenetV1/MobilenetV1/Conv2d_2_pointwise/BatchNorm/batchnorm/mul/eightbit, MobilenetV1/MobilenetV1/Conv2d_2_pointwise/BatchNorm/batchnorm/mul/eightbit:1, MobilenetV1/MobilenetV1/Conv2d_2_pointwise/BatchNorm/batchnorm/mul/eightbit:2, MobilenetV1/MobilenetV1/Conv2d_2_pointwise/BatchNorm/batchnorm/mul/eightbit/requant_range, MobilenetV1/MobilenetV1/Conv2d_2_pointwise/BatchNorm/batchnorm/mul/eightbit/requant_range:1)]]
native : benchmark_model.cc:269 Failed on run 0
native : benchmark_model.cc:452 Timing failed with Invalid argument: requested_output_max must be >= requested_output_min, but got nan and 0
	 [[Node: MobilenetV1/MobilenetV1/Conv2d_2_pointwise/BatchNorm/batchnorm/mul/eightbit/requantize = Requantize[Tinput=DT_QINT32, out_type=DT_QUINT8, _device="/job:localhost/replica:0/task:0/cpu:0"](MobilenetV1/MobilenetV1/Conv2d_2_pointwise/BatchNorm/batchnorm/mul/eightbit, MobilenetV1/MobilenetV1/Conv2d_2_pointwise/BatchNorm/batchnorm/mul/eightbit:1, MobilenetV1/MobilenetV1/Conv2d_2_pointwise/BatchNorm/batchnorm/mul/eightbit:2, MobilenetV1/MobilenetV1/Conv2d_2_pointwise/BatchNorm/batchnorm/mul/eightbit/requant_range, MobilenetV1/MobilenetV1/Conv2d_2_pointwise/BatchNorm/batchnorm/mul/eightbit/requant_range:1)]]
----

**Comment by snownus** [comment]
@qjivy  Have u addressed the problem?

**Comment by snownus** [comment]
@petewarden  I have came across similar problem. InvalidArgumentError (see above for traceback): requested_output_max must be >= requested_output_min, but got -33025.5117 and 0
	 [[Node: SecondStagePostprocessor/BatchMultiClassNonMaxSuppression/MultiClassNonMaxSuppression/ClipToWindow_2/Area/mul/eightbit/requantize = Requantize[Tinput=DT_QINT32, out_type=DT_QUINT8, _device="/job:localhost/replica:0/task:0/device:CPU:0"](SecondStagePostprocessor/BatchMultiClassNonMaxSuppression/MultiClassNonMaxSuppression/ClipToWindow_2/Area/mul/eightbit, SecondStagePostprocessor/BatchMultiClassNonMaxSuppression/MultiClassNonMaxSuppression/ClipToWindow_2/Area/mul/eightbit:1, SecondStagePostprocessor/BatchMultiClassNonMaxSuppression/MultiClassNonMaxSuppression/ClipToWindow_2/Area/mul/eightbit:2, SecondStagePostprocessor/BatchMultiClassNonMaxSuppression/MultiClassNonMaxSuppression/ClipToWindow_2/Area/mul/eightbit/requant_range, SecondStagePostprocessor/BatchMultiClassNonMaxSuppression/MultiClassNonMaxSuppression/ClipToWindow_2/Area/mul/eightbit/requant_range:1)]]
 

In Faster RCNN, can you help to fix this problem?

**Comment by qjivy** [comment]
@snownus, I still can't solve this problem at the moment.

**Comment by aselle** [comment]
@petewarden, can you think of anyone else to look at this?

**Comment by snownus** [comment]
@aselle , yes. we would appreciate it a lot if you can help to address the problem.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by aselle** [comment]
Ping @petewarden

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
A member of the TensorFlow organization has replied after the stat:awaiting tensorflower label was applied.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 112 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 127 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 142 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 157 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by petewarden** [comment]
I'm closing this now, since we've moved away from this code path in favor of TensorFlow Lite. Sorry for the long delay!
